Gems/ROS2/Code/Source/VehicleDynamics/Utilities.cpp

```diff
@@ -129,16 +148,46 @@ namespace ROS2::VehicleDynamics::Utilities
                         axle.m_axleTag.c_str());
                     continue;
                 }
-                AZ::Vector3 driveDir = controllerComponent->m_driveDir;
-                driveDir.Normalize();
+
+                PhysX::HingeJointComponent* hingeComponent{ nullptr };
+                hingeComponent = wheelEntity->FindComponent<PhysX::HingeJointComponent>();
+
+                if (!hingeComponent)
+                {
+                    AZ_Warning(
+                        "GetAllDriveWheelsData",
+                        false,
+                        "Wheel entity for axle %s is missing a HingeJointComponent component, ignoring",
+                        axle.m_axleTag.c_str());
+                    continue;
+                }
 
                 VehicleDynamics::WheelDynamicsData wheelData;
                 wheelData.m_wheelEntity = wheel;
-                wheelData.m_driveAxis = driveDir;
+                wheelData.m_hingeJoint = hingeComponent->GetId();
                 wheelData.m_wheelRadius = axle.m_wheelRadius;
                 driveWheelEntities.push_back(wheelData);
             }
         }
         return driveWheelEntities;
     }
+
+    AZ::EntityComponentIdPair GetWheelPhysxHinge(const AZ::EntityId entityId)
+    {
+        AZ::Entity* wheelEntity = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(wheelEntity, &AZ::ComponentApplicationRequests::FindEntity, entityId);
+        if (!wheelEntity){
+            AZ_Warning("GetWheelDynamicData", false , "Entity %s was not found", entityId.ToString().c_str() );
+            return  AZ::EntityComponentIdPair();
+        }
+        PhysX::HingeJointComponent* hingeComponent{ nullptr };
+        hingeComponent = wheelEntity->FindComponent<PhysX::HingeJointComponent>();
+        if (!hingeComponent){
+            AZ_Warning("GetWheelDynamicData", false , "Entity %s has no PhysX::HingeJointComponent", entityId.ToString().c_str() );
+            return  AZ::EntityComponentIdPair();
```

Comment:
nitpick: extra whitespace between `return` and `AZ::EntityComponentIdPair()`
```suggestion
            return AZ::EntityComponentIdPair();
```